nrcatalogtools/waveform.py

```diff
@@ -215,85 +215,11 @@ def load_from_targz(cls, file_path, metadata={}, verbosity=0):
-            cls._metadata = metadata
-
-        ell_min, ell_max = 99, -1
-        t_min, t_max, dt = -1e99, 1e99, 1
-
-        file_tag = get_tag(file_path)
-        mode_data = {}
-        reference_mode_num_for_length = ()
-        possible_ascii_extensions = ["asc", "dat", "txt"]
-
-        with tarfile.open(file_path, "r:gz") as tar:
-            for dat_file in tar.getmembers():
-                dat_file_name = dat_file.name
-                if file_tag not in dat_file_name or np.all(
-                    [
-                        f".{ext}" not in dat_file_name
-                        for ext in possible_ascii_extensions
-                    ]
-                ):
-                    continue
-                ell, em = get_el_em_from_filename(dat_file_name)
-                with tar.extractfile(dat_file_name) as f:
-                    reference_mode_num_for_length = (ell, em)
-                    mode_data[(ell, em)] = np.loadtxt(f)
-                    # mode_data[get_tag(dat_file_name)] = np.loadtxt(f)
-                # get the minimum time and maximum time stamps for all modes
-                t_min = max(t_min, mode_data[(ell, em)][0, 0])
-                t_max = min(t_max, mode_data[(ell, em)][-1, 0])
-                dt = min(
-                    dt,
-                    stat_mode(np.diff(mode_data[(ell, em)][:, 0]), keepdims=True)[0][0],
-                )
-                ell_min = min(ell_min, ell)
-                ell_max = max(ell_max, ell)
-
-        # We populate LM here because it has to be ordered, as the WaveformModes
-        # class expects an ordered data set.
-        LM = []
-        for ell in range(ELL_MIN, ELL_MAX + 1):
-            for em in range(-ell, ell + 1):
-                if (ell, em) in mode_data:
-                    LM.append([ell, em])
-                else:
-                    reference_mode = mode_data[reference_mode_num_for_length]
-                    mode_data[(ell, em)] = np.zeros(np.shape(reference_mode))
-                    LM.append([ell, em])
-
-        if len(LM) == 0:
-            raise RuntimeError(
-                "We did not find even one mode in the file. Perhaps the "
-                "format `amp_l?_m?` and `phase_l?_m?` is not the "
-                "nomenclature of datagroups in the input file?"
-            )
-
-        times = np.arange(t_min, t_max + 0.5 * dt, dt)
-        data = np.empty((len(times), len(LM)), dtype=complex)
-        for idx, (ell, em) in enumerate(LM):
-            mode_time, mode_real, mode_imag = mode_data[(ell, em)]
-            mode_real_interp = InterpolatedUnivariateSpline(mode_time, mode_real)
-            mode_imag_interp = InterpolatedUnivariateSpline(mode_time, mode_imag)
-            data[:, idx] = mode_real_interp(times) + 1j * mode_imag_interp(times)
+        wftools_modes_array = load_RIT_Psi4_data_from_disk(
+            data_file_path=file_path, output_modes_array=True, resam_type="finest"
+        )
```

Comment:
Lets **not** delegate this IO task to another package. Offloading minor math operations is fine, but this functionality is central to the `nr-catalog-tolls` package to begin with.